tengine/README.md

```diff
@@ -0,0 +1,345 @@
+        // does not have a trailing slash. Having a trailing slash in
+        // the views_path config would make Drake look for ...
+        //
+        //     ./views_path//index.html
+        //
+        // ... which would cause an error.
+        this.response.body = this.response.render(
+          "/index.html",
+          {
+            message: "Hella using Drake.",
+            template_engines: [
+              {
+                name: "dejs",
+                url: "https://github.com/syumai/dejs",
+              },
+              {
+                name: "Dinja",
+                url: "https://github.com/denjucks/dinja",
+              },
+              {
+                name: "Eta",
+                url: "https://github.com/eta-dev/eta",
+              }
+            ],
+          }
+        );
+
+        return this.response;
+      }
+    }
+    ```
+
+3. Create your `index.html` file.
+
+    ```html
+    <!DOCTYPE html>
+    <html class="h-full w-full">
+      <head>
+        <meta charset="utf-8"/>
+        <meta name="viewport" content="width=device-width, minimum-scale=1.0, user-scalable=no"/>
+        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/tailwindcss/dist/tailwind.min.css">
+        <title>Tengine</title>
+      </head>
+      <body style="background: #f4f4f4">
+        <div style="max-width: 640px; margin: 50px auto;">
+          <h1 class="text-5xl mb-5"><% message %></h1>
+          <p class="mb-5">Drake is a template engine. Some other template engines are:</p>
+          <% include_partial("/template_engines.html") %>
+        </div>
+      </body>
+    </html>
+    ```
+
+4. Create your `template_engines.html` file.
+
+    ```html
+    <ul class="list-disc ml-5">
+      <% for (const index in template_engines) { %>
+        <li>
+          <span class="text-bold"><% template_engines[index].name %>: </span>
+          <a href="<% template_engines[index].url %>" target="_BLANK"><% template_engines[index].url %></a>
+        </li>
+      <% } %>
+    </ul>
+    ```
+
+### Drake: Verification
+
+1. Run your `app.ts` file.
+
+    ```
+    deno run --allow-net --allow-read app.ts
+    ```
+
+2. Navigate to `localhost:1447` in your browser. You should see an HTML page with the following text:
+
+   ```text
+   Hella using Drake.
+   ```
+
+## Tutorial: Using Eta
+
+This tutorial teaches you how to use Eta.
+
+_Note: Eta uses unstable Deno APIs. Therefore, you must pass in the `--unstable` flag when you run your application._
+
+### Eta: Folder Structure End State
+
+```
+▾ /path/to/your/project/
+    ▾ views/
+        index.html
+    app.ts
+    home_resource.ts
+```
+
+### Eta: Steps
+
+1. Create your `app.ts` file.
+
+    ```typescript
+    import { Drash } from "https://deno.land/x/drash@v1.3.0/mod.ts";
+    import { HomeResource } from "./home_resource.ts";
+    import { renderFile, configure } from "https://deno.land/x/eta@v1.6.0/mod.ts"
```

Comment:
can this line bee added to the bumper ci process? so when the bumper ci updates dependencies in `deps.ts` and others, it can also re-write this file to fetch the latest version from denos cdn

Though we can probably use the services repo for this as well?

Reply:
resolving as i actually mentioned this in a better way in the main comment above